### PR TITLE
Added ts-loader example

### PR DESCRIPTION
**Pull request by johnnyreilly**
This should resolve https://github.com/amireh/happypack/issues/164

I'm afraid I can't run the test as I'm on Windows.  I've done my best to guess how it should work - may need your help there if it doesn't run.  Fingers crossed!

### CODE REVIEW

**Comment by amireh** [comment]
So responsive! Haha, thank you, I'll try it out as soon as I can.

**Comment by amireh** [comment]
Hmm, I'm seeing something strange where the fork-ts-checker plugin isn't despawning its services. Is this intended? The example fails to run because the process is never killed. Here's a snippet that shows the remaining (zombie) processes:

```
user:happypack$ ps -ef | grep ts-loader
user   18577     1 42 01:04 pts/19   00:00:00 ~/.nvm/versions/node/v6.3.1/bin/node --max-old-space-size=2048 ~/Workspace/Projects/happypack/examples/ts-loader--webpack2/node_modules/fork-ts-checker-webpack-plugin/lib/service.js
user   18583     1 44 01:04 pts/19   00:00:00 ~/.nvm/versions/node/v6.3.1/bin/node --max-old-space-size=2048 ~/Workspace/Projects/happypack/examples/ts-loader--webpack2/node_modules/fork-ts-checker-webpack-plugin/lib/service.js
user   18584     1 41 01:04 pts/19   00:00:00 ~/.nvm/versions/node/v6.3.1/bin/node --max-old-space-size=2048 ~/Workspace/Projects/happypack/examples/ts-loader--webpack2/node_modules/fork-ts-checker-webpack-plugin/lib/service.js
user   18586     1 41 01:04 pts/19   00:00:00 ~/.nvm/versions/node/v6.3.1/bin/node --max-old-space-size=2048 ~/Workspace/Projects/happypack/examples/ts-loader--webpack2/node_modules/fork-ts-checker-webpack-plugin/lib/service.js
user   18591     1 43 01:04 pts/19   00:00:00 ~/.nvm/versions/node/v6.3.1/bin/node --max-old-space-size=2048 ~/Workspace/Projects/happypack/examples/ts-loader--webpack2/node_modules/fork-ts-checker-webpack-plugin/lib/service.js
user   18602     1 43 01:04 pts/19   00:00:00 ~/.nvm/versions/node/v6.3.1/bin/node --max-old-space-size=2048 ~/Workspace/Projects/happypack/examples/ts-loader--webpack2/node_modules/fork-ts-checker-webpack-plugin/lib/service.js
user   18603     1 43 01:04 pts/19   00:00:00 ~/.nvm/versions/node/v6.3.1/bin/node --max-old-space-size=2048 ~/Workspace/Projects/happypack/examples/ts-loader--webpack2/node_modules/fork-ts-checker-webpack-plugin/lib/service.js
user   18614     1 39 01:04 pts/19   00:00:00 ~/.nvm/versions/node/v6.3.1/bin/node --max-old-space-size=2048 ~/Workspace/Projects/happypack/examples/ts-loader--webpack2/node_modules/fork-ts-checker-webpack-plugin/lib/service.js
user   18620     1 43 01:04 pts/19   00:00:00 ~/.nvm/versions/node/v6.3.1/bin/node --max-old-space-size=2048 ~/Workspace/Projects/happypack/examples/ts-loader--webpack2/node_modules/fork-ts-checker-webpack-plugin/lib/service.js
user   18626     1 78 01:04 pts/19   00:00:01 ~/.nvm/versions/node/v6.3.1/bin/node --max-old-space-size=2048 ~/Workspace/Projects/happypack/examples/ts-loader--webpack2/node_modules/fork-ts-checker-webpack-plugin/lib/service.js
```

I have to manually kill them to get this to clean up.

**Comment by johnnyreilly** [comment]
@pietr-oles can you comment on this please?

**Comment by johnnyreilly** [comment]
Let's try that again 😄 @piotr-oles would you be able to comment on this please?  It sounds like there might be an issue with fork-ts-checker plugin despawning its services...

**Comment by dziamid** [comment]
This works great except with dll plugin. It build the dll, but the process does not exit.
```

Building the Webpack DLL...
Loaded environment variables for development:
{ BASE_API: '//api.dev2.noviopus.com/api', PORT: '3000' }
version 1.7.2
Happy[ts]: Version: 4.0.0-beta.1. Threads: 4
Happy[ts]: All set; signaling webpack to proceed.
Starting type checking service...
Using 2 workers with 2048MB memory limit
/home/dziamid/dev/noviopus/tsconfig.json


No type errors found
Version: typescript 2.3.3
Time: 28720ms
chunk    {0} reactBoilerplateDeps.dll.js (reactBoilerplateDeps) 12.8 MB [entry] [rendered] 
//hangs here

```

**Comment by johnnyreilly** [comment]
Thanks @dziamid - that's good to know. Hopefully @piotr-oles can advise. I may open an issue on fork checker to track.

**Comment by piotr-oles** [comment]
The problem with fork-ts-checker-webpack-plugin was misleading configuration. I've released v0.2.0 with some improvements you can check in changelog. 

The main change is removing `blockEmit` option and setting it automatically (false for 'watch' mode, true for 'one build' mode). Also, it doesn't start chokidar watcher on build mode (the process didn't want to exit because of files watcher).

**Comment by amireh** [comment]
I have verified that it works once we upgrade to 0.2.0 - merging this! Thanks all.

**Comment by amireh** [comment]
The README now [indicates TypeScript compatibility](https://github.com/amireh/happypack#does-it-work-with-typescript) - thank you all for your work.

**Comment by johnnyreilly** [comment]
I've simplified the example in the PR and removed api bits that are no longer part of fork-ts-checker.  Hope that helps!

**Comment by piotr-oles** [comment]
I've forgot to submit revew yesterday, it was pending. Never mind, it's not up to date :)

File: examples/ts-loader--webpack2/webpack.config--raw.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+var process = require('process');
+var path = require('path');
+var ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
+
+module.exports = {
+    context: __dirname, // to automatically find tsconfig.json
+    entry: './src/index.ts',
+    output: {
+        path: path.resolve(__dirname, 'dist'),
+        filename: '[name].raw.js'
```
Comment (johnnyreilly) [inline examples/ts-loader--webpack2/webpack.config--raw.js]: oooh you might be right!

**Pull request merged**